lib/index.js

```diff
@@ -345,13 +345,20 @@ exports.deepEqual = function (obj, ref, seen) {
 
 // Remove duplicate items from array
 
-exports.unique = function (array, key) {
+exports.unique = function (array, keys) {
+
+    if (keys && !Array.isArray(keys)) {
+        keys = [keys];
+    }
 
     var index = {};
     var result = [];
 
     for (var i = 0, il = array.length; i < il; ++i) {
-        var id = (key ? array[i][key] : array[i]);
+        var id = (keys ? keys.reduce(function(id, key) {
```

Comment:
That makes sense that it is slower since it provides closure and handling of the loop. I'm more than glad to replace it with the for loop equivalent.

I wanted to get a clear picture of what the cost is in this case, here's some benchmarks on three methods.

```bash
  Hoek.uniqueReduce(array, key)  x 410,757 ops/sec ±1.35% (90 runs sampled)
  Hoek.uniqueLoop(array, key)    x 524,598 ops/sec ±1.84% (90 runs sampled)
  Hoek.uniqueForEach(array, key) x 430,065 ops/sec ±0.98% (94 runs sampled)

  Hoek.uniqueReduce(array, [keys])  x 211,107 ops/sec ±1.14% (94 runs sampled)
  Hoek.uniqueLoop(array, [keys])    x 262,689 ops/sec ±0.74% (97 runs sampled)
  Hoek.uniqueForEach(array, [keys]) x 199,105 ops/sec ±1.19% (93 runs sampled)
```

`uniqueLoop` is the standard `for (i = 0....)` implementation, `uniqueReduce` is the reduce implementation, and `uniqueForEach` is the implementation using `keys.forEach(function(key)...)`.

You can find the [benchmarks branch](https://github.com/DavidTPate/hoek/blob/3dd84a927c8956bfee0f3534fc5eba5ff29620e6/bench/bench.js) here if you want to run them yourself.